extensions/amp-story-360/0.1/amp-story-360.js

```diff
@@ -406,35 +417,45 @@ export class AmpStory360 extends AMP.BaseElement {
   }
 
   /**
-   * Creates a device orientation listener and sets gyroscopeControls_ state.
-   * If listener is not called in 1000ms, remove listener and resume animation.
-   * This happens on desktop browsers that support deviceorientation but aren't in motion.
+   * Listens for deviceorientation events.
+   *
+   * Some browsers support the 'deviceorientation' event but never call it.
+   * This waits for one call before initiating a constant listener.
    * @private
    */
   enableGyroscope_() {
-    this.gyroscopeControls_ = true;
-
-    const checkNoMotion = this.timer_.delay(() => {
-      this.gyroscopeControls_ = false;
-      if (this.isReady_ && this.isPlaying_) {
-        this.animate_();
-      }
-    }, 1000);
-
-    let rafTimeout;
-
-    this.win.addEventListener('deviceorientation', (e) => {
-      if (this.isReady_ && this.isOnActivePage_) {
-        // Debounce onDeviceOrientation_ to rAF.
-        rafTimeout && this.win.cancelAnimationFrame(rafTimeout);
-        rafTimeout = this.win.requestAnimationFrame(() =>
-          this.onDeviceOrientation_(e)
-        );
-      }
-      this.timer_.cancel(checkNoMotion);
+    // Listen for one call before initiating.
+    listenOncePromise(this.win, 'deviceorientation').then(() => {
+      this.gyroscopeControls_ = true;
+      // Debounce onDeviceOrientation_ to rAF.
+      let rafTimeout;
+      this.win.addEventListener('deviceorientation', (e) => {
+        if (this.isReady_ && this.isOnActivePage_) {
+          rafTimeout && this.win.cancelAnimationFrame(rafTimeout);
+          rafTimeout = this.win.requestAnimationFrame(() =>
+            this.onDeviceOrientation_(e)
+          );
+        }
+      });
+      this.maybeShowDiscoveryAnimation_();
     });
   }
 
+  /** @private */
+  maybeShowDiscoveryAnimation_() {
+    if (
+      this.isOnActivePage_ &&
+      this.gyroscopeControls_ &&
+      !this.element.ownerDocument.querySelector(
+        '.i-amphtml-story-360-discovery'
```

Comment:
Please add a comment in the `buildDiscoveryTemplate` so we remember we need to change this class in case of update

Reply:
can't we just use a const here and in the template ? :)

Reply:
> can't we just use a const here and in the template ? :)

Do you mean check if `this.discoveryTemplate_`? That would only check this instance of the component.
I'm not sure I follow the question.

Reply:
> can't we just use a const here and in the template ? :)

That's a good idea but `htmlFor()` Templates can't contain expressions. 
https://travis-ci.org/github/ampproject/amphtml/jobs/732973339#L608

I'll add a comment in `buildDiscoveryTemplate` as Gabriel suggests.